pkg/services/authz/rbac/service.go

```diff
@@ -374,3 +488,16 @@ func (s *Service) buildFolderTree(ctx context.Context, ns claims.NamespaceInfo)
 
 	return folderMap, nil
 }
+
+func getChildren(folderMap map[string]FolderNode, folderUID string) []string {
```

Comment:
We could probably pass in the set to save some memory allocation and avoid a couple of loops in both the calling function and in this function.
```suggestion
func getChildren(folderMap map[string][]string, identifier string, folderSet map[string]struct{}) {
```